### ISSUE TITLE
lin_comb method does not produce symmetric Hessian

### ISSUE THREAD

**Issue by molar-volume** [Qiskit/qiskit-terra#5566]
### Information

- **Qiskit Terra version**: 0.16.1     698bd3faf546ac335c3e1a6b202a089e75a19fcd
- **Python version**: 3.7
- **Operating system**: OSX

### What is the current behavior?
Methods `param_shift` and `fin_diff` produce symmetric hessian, but `lin_comb` does not.

### Steps to reproduce the problem
```python
import numpy as np
from qiskit import QuantumCircuit, QuantumRegister
from qiskit.circuit import Parameter
from qiskit.opflow import X, Z, StateFn, CircuitStateFn
from qiskit.opflow.gradients import Hessian

ham = 0.5 * X - 1 * Z
a = Parameter('a')
b = Parameter('b')
params = [(a, b), (b, a)]

q = QuantumRegister(1)
qc = QuantumCircuit(q)
qc.h(q)
qc.rz(a, q[0])
qc.rx(b, q[0])

op = ~StateFn(ham) @ CircuitStateFn(primitive=qc, coeff=1.)
state_hess = Hessian(hess_method='lin_comb').convert(operator=op, params=params)

print(state_hess.assign_parameters({a: np.pi / 2, b: np.pi / 4}).eval())  # [0j, (-0.7071067811865474-7.8e-16j)]
```

### What is the expected behavior?
```python
print(state_hess.assign_parameters({a: np.pi / 2, b: np.pi / 4}).eval())  # [0j, 0j]
```

**Comment by Cryoris** [comment]
This will be fixed by #5501

**Comment by molar-volume** [comment]
fixed by #5501
